app/models/deployment.rb

```diff
@@ -0,0 +1,30 @@
+class Deployment < ApplicationRecord
+  validates :name,
+    presence: true,
+    uniqueness: { case_sensitive: false },
+    format: { with: NAME_FORMAT },
+    length: { minimum: 1, maximum: 255 }
+
+  belongs_to :cluster
+
+  def container_names
+    (0..count-1).map{ |i| "#{name}-%02d" % (i+1) }
+  end
+
+  def managed_containers
```

Comment:
why do we need this function? 
why does deployment does not have `has_many` relation to container?

Reply:
We have these functions to help us managed existing containers in PF-mono. Currently, we have whole picture of provisioning and bootstrapping in BM and PF-mono only have containers information. We try to find a way how to move whole picture of provisioning and bootstrapping in BM to PF-mono.

After we have that in pf-mono, we can refactor these functions and create a migration script for mapping and set deployment `has_many` relation to container.

Reply:
ok, then this is potentially method that will be deprecated.
Please give comments on all methods / APIs that eventually will be deprecated

Reply:
Noted, Mas. We will also have deployment notes for this.